src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -3700,9 +3701,21 @@ _public_ int sd_varlink_server_listen_auto(sd_varlink_server *s) {
                 n++;
         }
 
+        return n;
+}
+
+_public_ int sd_varlink_server_listen_auto(sd_varlink_server *s) {
+        int n;
+
+        assert_return(s, -EINVAL);
+
+        n = sd_varlink_server_listen_name(s, "varlink");
+
         /* Let's listen on an explicitly specified address */
         const char *e = secure_getenv("SYSTEMD_VARLINK_LISTEN");
         if (e) {
+                int r;
```

Comment:
According to our coding style, r should always be declared at the very beginning of the function